sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -311,6 +311,67 @@ def predict(self, X):
         return X[:, 0]
 
 
+class OneClassSampleErrorClassifier(BaseBadClassifier):
+    def fit(self, X, y, sample_weight=None):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+
+        cls, y = np.unique(y, return_inverse=True)
+        nb_cls = cls.shape[0]
+        if nb_cls < 2:
+            raise ValueError("normal class error")
+
+        # find the number of class after trimming
+        if sample_weight is not None:
+            nb_cls = cls.shape[0]
+            if isinstance(sample_weight, np.ndarray) and len(sample_weight) > 0:
+                nb_cls = np.count_nonzero(np.bincount(y, sample_weight))
+            if nb_cls < 2:
+                raise ValueError("Nonsensical Error")
+
+        return self
+
+    def predict(self, X):
+        X = check_array(X)
+        return np.ones(X.shape[0])
+
+
+class OneClassSampleErrorClassifierPredict(BaseBadClassifier):
+    def __init__(self, flag=False):
+        self.flag = flag
+
+    def fit(self, X, y, sample_weight=None):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+
+        cls, y = np.unique(y, return_inverse=True)
+        nb_cls = cls.shape[0]
+        if nb_cls < 2:
+            raise ValueError("normal class error")
+
+        # find the number of class after trimming
+        if sample_weight is not None:
+            nb_cls = cls.shape[0]
+            if isinstance(sample_weight, np.ndarray) and len(sample_weight) > 0:
+                nb_cls = np.count_nonzero(np.bincount(y, sample_weight))
+            if nb_cls < 2:
+                self.flag = True
+
+        self.flag_ = self.flag
```

Comment:
The "flag" is very generic. May we have a better name here?

```suggestion

        # find the number of class after trimming
        self.has_single_class_ = False
        if sample_weight is not None:
            nb_cls = cls.shape[0]
            if isinstance(sample_weight, np.ndarray) and len(sample_weight) > 0:
                nb_cls = np.count_nonzero(np.bincount(y, sample_weight))
            if nb_cls < 2:
                self.has_single_class_ = True
```

This means `predict` can be:

```python
def predict(self, X):
    if self.has_single_class_:
        return np.zeros(...)
    ...
```

This also means we do not need `self.flag` anymore.